### ISSUE TITLE
Accelerometer and gyroscope don't work with Switch Pro Controller on Windows 10

### ISSUE THREAD

**Issue by Kilicool64** [libretro/RetroArch#15196]
I can't get motion controls to work in Kirby Tilt 'n' Tumble and WarioWare Twisted using mGBA. I tested the latter with VBA-M as well with no difference, so I don't think it's exclusive to one core.

It's the same issue as #15166, just with a different controller and OS, so perhaps they have the same cause. And I can also use this controller just fine with the standalone version of mGBA.

**Comment by Maxkatsur** [comment]
Why are they ignoring this issue is anyone's guess; I was trying to get my Logitech gamepad's analog stick use the emulated Gyroscope, but it doesn't work. Emulator: Sameboy (0.16.5)

**Comment by gouchi** [comment]
@Kilicool64 it should work now using SDL2 input/joypad driver using stable version.

I confirm it is working using DS4 on Linux with mGBA Libretro core.

Thank you.

**Comment by gouchi** [comment]
Hi @bcat, since your [update](https://github.com/libretro/RetroArch/pull/18635) and message related to Switch Pro controller.

You confirm it is working well now ? Provided that we update the SDL2 dll ?

Thank you for your help.

**Comment by bcat** [comment]
> Hi [@bcat](https://github.com/bcat), since your [update](https://github.com/libretro/RetroArch/pull/18635) and message related to Switch Pro controller.
> 
> You confirm it is working well now ? Provided that we update the SDL2 dll ?
> 
> Thank you for your help.

Oh, fascinating. I had no idea RetroArch had gyro support! Anyway, things are quite close to "just working", with one weird bug: the X axis (left / right) is flipped.

Specifically, I did the following:

* Installed a RetroArch Windows nightly. (Not strictly needed for the gyro to work, but I wanted the fix from #18635 so autoconfigs worked properly.)
* Replaced the bundled `SDL2.dll` file with the latest Windows build from the SDL project.
* Started Kirby Tilt n Tumble in the mGBA core.

Gyro worked out of the box with both a real Pro Controller and a third-party one (8BitDo Ultimate 2 Wireless put in "Switch mode" by pressing Home+Y to wake it up). However, the X axis is flipped with both controllers, as mentioned above.

I'm not sure if that's a bug in mGBA, RetroArch, or SDL. Do you know of any other cores / games I can test gyro in?

**Comment by gouchi** [comment]
@bcat 

> I'm not sure if that's a bug in mGBA, RetroArch, or SDL. Do you know of any other cores / games I can test gyro in?

If I understand correctly, [Remote RetroPad](https://github.com/libretro/RetroArch/blob/a458d7e748c6f42b0b6ac8b5a21d06a045f71fa9/cores/libretro-net-retropad/net_retropad_core.c#L450) supports it  via Main > Load Core > Start Remote RetroPad. And in the Core options , select Screen > Sensor ?

Thank you.

**Comment by bcat** [comment]
Hmm, nothing actually updates on that screen of the Remote RetroPad. It shows a bunch of sensors, but moving the controller around doesn't affect the display at all.

**Comment by bcat** [comment]
Okay, I tried Remote RetroPad it again and this time it updated (?). I'm not entirely sure how to "read" the screen, but it _looks_ like the tilt movement is being reported in the correct direction, as far as I can tell.

**Comment by gouchi** [comment]
@Kilicool64 @bcat I made a test on Windows 10 with DS4 and SDL2 driver and I confirm gyro is working.

So it should work with the Switch Pro controller also.

**Comment by gouchi** [comment]
@Kilicool64 @bcat 

It seems we are shipping [SD2.dll](https://www.virustotal.com/gui/file/02859fa51aa91ea975e641a0da13c0afaaab7e87f11895d46b420cc5856fae77/details) version [2.0.14](https://github.com/libsdl-org/SDL/releases/tag/release-2.0.14) which is from 2021.

But gyro support for Switch Pro controller has been introduced later on with this [commit](https://github.com/libsdl-org/SDL/commit/4aec54988ca71f30ebf9fd5c37821f3d7e103aa5) in 2023.

So you will have to update `SDL2.dll` in RetroArch directory, using for example [2.32.10](https://github.com/libsdl-org/SDL/releases/tag/release-2.32.10).

And it should work.

**Comment by bcat** [comment]
Indeed, I have been using an `SDL2.dll` I swapped to 2.32.10 for some months now, and it works quite well. (I had to upgrade to get better support for e.g. the NSO N64 controller.)

**Comment by Kilicool64** [comment]
It'll probably be a long time before I try playing these games again (and if I do, I don't know if I'll use RetroArch), but I'll take your word that updating SDL2 solves the issue.

**Comment by Starman99x** [comment]
> [@Kilicool64](https://github.com/Kilicool64) [@bcat](https://github.com/bcat)
> 
> It seems we are shipping [SD2.dll](https://www.virustotal.com/gui/file/02859fa51aa91ea975e641a0da13c0afaaab7e87f11895d46b420cc5856fae77/details) version [2.0.14](https://github.com/libsdl-org/SDL/releases/tag/release-2.0.14) which is from 2021.
> 
> But gyro support for Switch Pro controller has been introduced later on with this [commit](https://github.com/libsdl-org/SDL/commit/4aec54988ca71f30ebf9fd5c37821f3d7e103aa5) in 2023.
> 
> So you will have to update `SDL2.dll` in RetroArch directory, using for example [2.32.10](https://github.com/libsdl-org/SDL/releases/tag/release-2.32.10).
> 
> And it should work.

Yes, this finally worked for me. I hope SDL2.dll will be updated in RetroArch soon.

**Comment by gouchi** [comment]
[Linked issue](https://github.com/libretro/RetroArch/issues/16717).

**Comment by allanlc11** [comment]
> Gyro worked out of the box with both a real Pro Controller and a third-party one (8BitDo Ultimate 2 Wireless put in "Switch mode" by pressing Home+Y to wake it up). However, the X axis is flipped with both controllers, as mentioned above.
> 
> I'm not sure if that's a bug in mGBA, RetroArch, or SDL. Do you know of any other cores / games I can test gyro in?

Can confirm the same issue. It works out of the box but the X axis is flipped.

@bcat  You can test this on the Azahar core. The N64 Zelda games had motion controls added for the camera movement. It's correct on standalone Azahar and mGBA (and Cemu) but inverted on Retroarch.

You can flip the axis by setting sensitivity to negative numbers, but then the other axis are flipped...
